### PR TITLE
Angular 11

### PR DESCRIPTION
**Pull request by sulimani55**

### CODE REVIEW

**Review by tltk90 — CHANGES_REQUESTED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -1,41 +1,48 @@
-/*
```
[the remark this answers is not in the document]
Comment (sulimani55) [reply tsconfig.json:1]: it did it automatically, its fixed now

File: src/app/@ansyn/ol/plugins/annotations/annotations-context-menu/models/multi-choice-attribute.ts [hunk 2]
```diff
@@ -7,15 +7,6 @@ export class MultiChoiceAttribute extends AttributeBase<IKeyValuePair<string>[]>
 	private selectedOptions: IKeyValuePair<string>[];
```
[the remark this answers is not in the document]
Comment (sulimani55) [reply src/app/@ansyn/ol/plugins/annotations/annotations-context-menu/models/multi-choice-attribute.ts:7]: i added a get and set method for this attribute

File: e2e/tsconfig.e2e.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "../tsconfig.base.json",
+  "extends": "../tsconfig.json",
```
[the remark this answers is not in the document]
Comment (sulimani55) [reply e2e/tsconfig.e2e.json:2]: fixed

File: src/app/@ansyn/ansyn/modules/core/services/storage/storage.service.ts [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+import { Observable } from 'rxjs';
 import { ICoreConfig } from '../../models/core.config.model';
 import { CoreConfig } from '../../models/core.config';
-import { FetchService } from '../fetch.service';
 
 export interface IEntity {
 	creationTime: Date;
 	id: string;
+	name?: string;
+	autoSave?: boolean;
```
[the remark this answers is not in the document]
Comment (sulimani55) [reply src/app/@ansyn/ansyn/modules/core/services/storage/storage.service.ts:11]: isnt there a whole pr to remove autosave completely? because its still shows in several sections in the code

File: src/app/@ansyn/ansyn/components/ansyn-footer/ansyn-footer.component.spec.ts [hunk 5]
```diff
@@ -19,10 +19,11 @@ describe('AnsynFooterComponent', () => {
 		selector: 'ansyn-overlay-navigation-bar'
 	});
 
+	beforeEach(waitForAsync(() => {
```
Comment (ramtob) [inline src/app/@ansyn/ansyn/components/ansyn-footer/ansyn-footer.component.spec.ts:22]: Is this addition needed?

File: src/app/@ansyn/ansyn/modules/menu-items/cases/models/case.model.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import { IDilutedOverlay, IDilutedOverlaysHash, IOverlay, IOverlaysHash } from '../../../overlays/models/overlay.model';
-import { Feature, MultiPolygon, Point, Polygon } from 'geojson';
+import { Feature, MultiPolygon, Point, Polygon, Position } from 'geojson';
```
Comment (ramtob) [inline src/app/@ansyn/ansyn/modules/menu-items/cases/models/case.model.ts:2]: Is  this addition needed?

File: src/app/@ansyn/ansyn/modules/overlays/components/overlay-timeline-status/overlay-timeline-status.component.spec.ts [hunk 7]
```diff
@@ -47,7 +47,7 @@ describe('OverlayTimelineStatusComponent', () => {
 		expect(component).toBeTruthy();
 	});
 
-	it('Should change "showStatus" to true only if statusMessage exists and not loading', async(async () => {
+	it('Should change "showStatus" to true only if statusMessage exists and not loading', waitForAsync(async () => {
```
Comment (ramtob) [inline src/app/@ansyn/ansyn/modules/overlays/components/overlay-timeline-status/overlay-timeline-status.component.spec.ts:50]: Seems like duplicate call? async(async(...))

File: src/app/@ansyn/map-facade/components/map-search-box/map-search-box.component.less [hunk 8]
```diff
@@ -5,7 +5,7 @@
 
 :host {
 	position: absolute;
-	top: @compass-top + @compass-width / 2 + 3; // middle of compass
+	top: calc(@compass-top + @compass-width / 2 + 3px); // middle of compass
```
Comment (ramtob) [inline src/app/@ansyn/map-facade/components/map-search-box/map-search-box.component.less:8]: Why this change?

(I see that the "left" definition below is similar, and you didn't change it)

File: src/app/app/app-routing.module.ts [hunk 9]
```diff
@@ -37,7 +37,7 @@ export const routes: Routes = [
 ];
 
 @NgModule({
-	imports: [RouterModule.forRoot(routes, { useHash: true })],
+	imports: [RouterModule.forRoot(routes, { useHash: true, relativeLinkResolution: 'legacy' })],
```
Comment (ramtob) [inline src/app/app/app-routing.module.ts:40]: Is this addition needed?

File: tsconfig.base.json [hunk 10]
```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "baseUrl": "./",
     "module": "es2020",
+    "skipLibCheck": true,
```
Comment (ramtob) [inline tsconfig.base.json:6]: Was this definition added automatically?

**Review by ramtob — CHANGES_REQUESTED** [review]
Very nice

File: src/app/@ansyn/ansyn/components/ansyn-footer/ansyn-footer.component.spec.ts (hunk 5 above)
Comment (sulimani55) [reply src/app/@ansyn/ansyn/components/ansyn-footer/ansyn-footer.component.spec.ts:22]: in angular11, the previous "async" method is redundent that is for not to confuse with the general "async" for js

File: tsconfig.base.json (hunk 10 above)
Comment (sulimani55) [reply tsconfig.base.json:6]: I added this definition because of a type error in @turf/turf version 6.
therefore, i added this variable, that ignores type errors in our node_modules

File: src/app/@ansyn/ansyn/components/ansyn-footer/ansyn-footer.component.spec.ts (hunk 5 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/components/ansyn-footer/ansyn-footer.component.spec.ts:22]: I am asking about something else. Why did you add `beforeEach`? It does not seem required.

File: tsconfig.base.json (hunk 10 above)
Comment (ramtob) [reply tsconfig.base.json:6]: Ok, this is new, I'm not sure how this will affect us, but we'll see.

Please answer all my other comments too.

File: src/app/@ansyn/ansyn/components/ansyn-footer/ansyn-footer.component.spec.ts (hunk 5 above)
Comment (sulimani55) [reply src/app/@ansyn/ansyn/components/ansyn-footer/ansyn-footer.component.spec.ts:22]: I guess it happened automatically, its fixed now thanks :)

File: src/app/@ansyn/ansyn/modules/menu-items/cases/models/case.model.ts (hunk 6 above)
Comment (sulimani55) [reply src/app/@ansyn/ansyn/modules/menu-items/cases/models/case.model.ts:2]: Apperantly we used `Position`

File: src/app/@ansyn/ansyn/modules/overlays/components/overlay-timeline-status/overlay-timeline-status.component.spec.ts (hunk 7 above)
Comment (sulimani55) [reply src/app/@ansyn/ansyn/modules/overlays/components/overlay-timeline-status/overlay-timeline-status.component.spec.ts:50]: the first function is the `waitForAsync`, which is the replacement for the previous `async` from angular.
the second `async` is the base js async

File: src/app/@ansyn/map-facade/components/map-search-box/map-search-box.component.less (hunk 8 above)
Comment (sulimani55) [reply src/app/@ansyn/map-facade/components/map-search-box/map-search-box.component.less:8]: `calc` no longer accepts arguments without unit (so it needs 3px instead of 3)

the "left" definition below doesn't use the `calc` function therefore it's capable of dealing without units

File: src/app/app/app-routing.module.ts (hunk 9 above)
Comment (sulimani55) [reply src/app/app/app-routing.module.ts:40]: It's not needed, it uses to access child routes better but there is no use for it in out project.
if you'd like to read about it it's here on the "Routes" section 
https://blog.ninja-squad.com/2020/11/11/what-is-new-angular-11.0/

File: src/app/@ansyn/map-facade/components/map-search-box/map-search-box.component.less (hunk 8 above)
Comment (ramtob) [reply src/app/@ansyn/map-facade/components/map-search-box/map-search-box.component.less:8]: But there was no `calc` before. You added the `calc`. Why did you add it?

File: src/app/app/app-routing.module.ts (hunk 9 above)
Comment (ramtob) [reply src/app/app/app-routing.module.ts]: Yes I read about it. But if it's not needed, why is it in the code? Why not remove it from the code, and then it will give the default value, which (as you say) is ok for us.

Comment (sulimani55) [reply src/app/app/app-routing.module.ts]: it's not longer in the code

File: src/app/@ansyn/map-facade/components/map-search-box/map-search-box.component.less (hunk 8 above)
Comment (sulimani55) [reply src/app/@ansyn/map-facade/components/map-search-box/map-search-box.component.less]: fixed

**Review by ramtob — APPROVED** [review]
🥇

**Review by tltk90 — APPROVED** [review]
